### PR TITLE
Make heartbeat more robust/less error prone

### PR DESCRIPTION
**Pull request by simonrondelez**

### CODE REVIEW

File: docker-heartbeat.sh [hunk 1]
```diff
@@ -53,15 +62,19 @@ test_docker_is_healthy() {
...
+    while [ $COUNTER -lt 5 ]; do
       dockerinfo=$(docker info)
       if [ -z "$dockerinfo" ] && [ $COUNTER -eq 4 ]; then
           handle_docker_not_running
       elif [ -n "$dockerinfo" ]; then
           exit;
+      elif [ $COUNTER -eq 4 ]; then
+        #something is still wrong
+        handle_docker_not_running;
+      fi
```
Comment (SamClinckspoor) [inline docker-heartbeat.sh]: one fi too many

File: docker-heartbeat.sh [hunk 2]
```diff
@@ -23,6 +23,12 @@ send_message() {
 
 handle_docker_not_running() {
     send_message "NOT_RUNNING" "Docker is not running"
+    docker_status=$(service docker restart)
+    EC="$?"
+    if [ $EC != 0 ]; then
+      #Docker restart did not help
+      handle_docker_not_healthy;
+    fi
```
Comment (SamClinckspoor) [inline docker-heartbeat.sh]: do we need a sleep() here?

Comment (simonrondelez) [inline docker-heartbeat.sh]: there is no sleep there as far as i can see

**Comment by iuriaranda** [comment]
I'm testing the script on a local vagrantbox, and I have a question: why is there a `test_docker_is_healthy` and a `test_docker_is_running`? What's the difference?

I say that because if docker is not running, it will be handled by the `handle_docker_not_healthy` function and restart the instance, without trying to restart the docker service first.

File: docker-heartbeat.sh [hunk 3]
```diff
@@ -38,13 +44,16 @@ handle_docker_not_healthy() {
 test_docker_is_healthy() {
     local dockertimeout
     COUNTER=0
-    while [[ $COUNTER < 5 ]]; do
+    while [ $COUNTER -lt 5 ]; do
       dockertimeout=$(eval "timeout ${timeout} docker info")
       EC="$?"
-      if [[ $EC != 0 && $COUNTER == 4 ]]; then
+      if [ $EC != 0 ] && [ $COUNTER -eq 4 ]; then
```
Comment (simonrondelez) [inline docker-heartbeat.sh]: this should be an || i think

**Pull request merged**